### ISSUE TITLE
Bug or Intender: OpaqueUserId is _not_ UuserID when the broadcaster + extension uses ID Linking

### ISSUE THREAD

**Issue by BarryCarlyon** [twitchdev/issues#559]
**Brief description**

as per

https://dev.twitch.tv/docs/extensions/reference#jwt-schema

> Broadcasters’ tokens are set with "U" + their Twitch user IDs, to avoid confusing opaque IDs with user IDs when the broadcaster is a viewer.

This doesn't seem to be the case

**How to reproduce**

In an extension that has requestID sharing off

calling `window.Twitch.ext.viewer.opaqueId` results in `U15185913` as expected

In an extension that has requestID sharing enabled and in use

calling `window.Twitch.ext.viewer.opaqueId` results in `Ugibberish` not as expected according to the docs.

**Expected behavior**

Not sure it inteneded or a bug. I'm erring towards bug.

**Comment by mgeyer** [comment]
Hi Barry, I'm a dev on the extensions team and tried to reproduce this issue, but wasn't able to. Can you provide more details on the steps you took to produce the bug?

The steps I took:
- Created an extension that ran the following javascript:
```
window.Twitch.ext.onAuthorized(function(auth) {
    console.log('The Extension JWT is ', auth.token);
    console.log('The user opaque id is ', window.Twitch.ext.viewer.opaqueId);
});
```
- Set `Request Identity Link` to `No, I would not like my extension to request an identity link.`
- Put extension into hosted test.
- Install the extension on my own channel
- Load the channel (logged in as the same user as the channel I installed the extension on)
- Check console and see `U123456789`
- Take extension out of hosted test
- Set `Request Identity Link` to `Yes, I would like my extension to request an identity link.`
- Move extension back to hosted test.
- Make sure that my account is disconnected from the extension in https://www.twitch.tv/settings/connections
- Refresh my channel (still logged in as my user)
- Check console and see `U123456789`

**Comment by BarryCarlyon** [comment]
![image](https://user-images.githubusercontent.com/20999/152854317-9ceeb100-d2ce-457b-b133-a4078c120c38.png)

**Comment by BarryCarlyon** [comment]
TLDR: exactly what you did.

**Comment by mgeyer** [comment]
Hmm that is strange. Ok so let me ask some questions (apologies for some that may seem obvious)
- The user you're logged in as is the same as the channel?
- Does the opaque ID in your extension JWT match the userID in the console?
- Have you tried installing the extension as a component and does it produce the same results?

**Comment by BarryCarlyon** [comment]
> The user you're logged in as is the same as the channel?

Yes hence "Edit Panels" and no visible dude icon in the Footer bar

> Does the opaque ID in your extension JWT match the userID in the console?

Yes

![image](https://user-images.githubusercontent.com/20999/152856352-ed5a0457-7960-4be6-8db2-bd28ae41a9a2.png)

> Have you tried installing the extension as a component and does it produce the same results?

No. I would have to go live.

**Comment by BarryCarlyon** [comment]
It also does it in the config view

![image](https://user-images.githubusercontent.com/20999/152857362-3270885a-b2d3-4470-af5c-63eb85ec5c01.png)

the test code https://github.com/barrycarlyon/twitch_extension_debug

It's in "local test" where the Testing base URI is the GitHub pages for this code

**Comment by mgeyer** [comment]
I've tried my own extension in both panel and config view and still see the correct userID. I get the same results if I'm on a different user as well. It could be something specific to your user. Can you try on a different account and see if the issue persists?

**Comment by BarryCarlyon** [comment]
I'll have a poke about further when I get a moment

**Comment by mgeyer** [comment]
Hey Barry, just wanted to check in again to see if you've had time to look at the issue.

**Comment by BarryCarlyon** [comment]
Here you go @mgeyer sorry for the delay:

As a componenet

![image](https://user-images.githubusercontent.com/20999/155743838-15a9a691-eea1-4cda-924f-15683935ecc0.png)

And as an overlay

![image](https://user-images.githubusercontent.com/20999/155743979-b969ed87-cdb4-44f1-9c89-3c62e6e03883.png)

Both return an "incorrect" opaque user ID

**Comment by mgeyer** [comment]
Thanks for the followup! Could you try to go to https://www.twitch.tv/settings/connections and see whether you have your extension in the list of Extension Connections? If so could you try to unlink and then refresh the page?

**Comment by jbulava** [comment]
@mgeyer created a ticket for this on ANC-2991.
